### PR TITLE
added 'limit', 'offset' and 'contains' query params for GET

### PR DESCRIPTION
**Pull request by MariaTuruta**

### CODE REVIEW

**Comment by MariaTuruta** [comment]
only works with groups, specialities and courses

File: src/routes/route-factory.js [hunk 1]
```diff
@@ -27,8 +28,20 @@ module.exports = function createRoute(repository) {
       .catch(next);
   }
 
+  function parseQueryParams(request, response, next) {
```
Comment (alxolr) [inline src/routes/route-factory.js]: This function is a middleware you can add it in the `src/middlewares` folder , please add unit tests also

File: src/repositories/repository-factory.js [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+const Sequelize = require('sequelize');
+
+const { Op } = Sequelize;
+
 module.exports = function createRepository(model) {
-  function list() {
-    return model.findAll({});
+  function list(queryParams) {
+    const { limit, offset, contains } = queryParams;
+    return model.findAll({
```
Comment (alxolr) [inline src/repositories/repository-factory.js]: In order to have a proper pagination the result from server should contain the requested params also:

```
{
  total: // total elements,
  limit: // limit asked by the requester
  offset: // offset asked by requester
  data: [] // the list of results
}
```
```

File: src/repositories/repository-factory.js [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+      offset,
+      limit,
+      where: {
+        name: {
+          [Op.like]: [`%${contains}%`],
```
Comment (alxolr) [inline src/repositories/repository-factory.js]: In order for Elena to test your route you should update the swagger documentation to include the limit, offset, and contains fields

Comment (MariaTuruta) [reply src/repositories/repository-factory.js]: where is swagger documentation? :)))

File: src/middlewares/parse-query.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = function parseQueryParams(request, response, next) {
+  if (!request.query.limit) request.query.limit = '50';
+  request.query.limit = parseInt(request.query.limit, 10);
+
```
Comment (ramzes13) [inline src/middlewares/parse-query.js]: you can add extra validation for NaN result like 
`request.query.limit = parseInt(request.query.limit) | 50;`

File: src/routes/route-factory.js [hunk 5]
```diff
@@ -28,9 +29,14 @@ module.exports = function createRoute(repository) {
   }
 
   function list(request, response, next) {
-    repository.list()
+    repository.list(request.query)
       .then((result) => {
-        response.json(result);
+        response.json({
+          total: result.length,
```
Comment (ramzes13) [inline src/routes/route-factory.js]: `result.length` may not be equal to `total`, especially when you set the limit and offset in sql query

Comment (MariaTuruta) [reply src/routes/route-factory.js]: 'limit' and 'offset' are query params from GET that are passed to the database

Comment (MariaTuruta) [reply src/routes/route-factory.js]: we get 'result' from database, why would its length not be equal to the total of items that are returned?

**Pull request closed without merging**